src/palace/manager/celery/tasks/work.py

```diff
@@ -44,16 +43,24 @@ def calculate_work_presentations(
                     ),
                 ):
                     for wp in work_policies:
-                        work = get_one(session, Work, id=wp.work_id)
-                        if not work:
-                            task.log.warning(
-                                f"No work with id={wp.work_id}. Skipping..."
-                            )
-                            continue
-                        work.calculate_presentation(policy=wp.policy)
+                        with task.transaction() as tx:
+                            work = get_one(tx, Work, id=wp.work_id)
+                            if not work:
+                                task.log.warning(
+                                    f"No work with id={wp.work_id}. Skipping..."
+                                )
+                                continue
+                            work.calculate_presentation(policy=wp.policy)
+                            remaining_work_policies.remove(wp)
             except Exception as e:
                 # if a failure occurs requeue the items so that can be recalculated in the next round
-                waiting.add(*work_policies)
+                task.log.error(
+                    f"Failed to calculate_presentation for {wp.work_id}. "
+                    f"Re-queuing remaining {len(remaining_work_policies)} of {len(work_policies)} "
+                    f"for future processing.",
+                    exc_info=e,
+                )
+                waiting.add(*remaining_work_policies)
```

Comment:
Minor: You may want to change this block to look like this:

```python
  except:
      # if a failure occurs requeue the items so that can be recalculated in the next round
      task.log.exception(
          f"Failed to calculate_presentation for {wp.work_id}. "
          f"Re-queuing remaining {len(remaining_work_policies)} of {len(work_policies)} "
          f"for future processing.",
      )
      waiting.add(*remaining_work_policies)
      raise
```

By using `raise` instead of `raise e` the exception will be linked to the line where it originally occurred, instead of being raised from this line.